controllers/Transactions/transactionSearchController.js

```diff
@@ -1,23 +1,75 @@
-const createHttpError = require('http-errors')
-const { Transaction } = require('../../database/models')
-const { endpointResponse } = require('../../helpers/success')
-const { catchAsync } = require('../../helpers/catchAsync')
+const createHttpError = require("http-errors");
+const { Transaction } = require("../../database/models");
+const { endpointResponse } = require("../../helpers/success");
+const { catchAsync } = require("../../helpers/catchAsync");
+const { ErrorObject } = require("../../helpers/error");
+const { Op } = require("sequelize");
+const { paginate } = require("../../middlewares/paginateMiddleware");
 
 module.exports = {
-  getAllTransactions: catchAsync(async (req, res, next) => {
+  listTransactions: catchAsync(async (req, res, next) => {
     try {
-      const response = await Transaction.findAll();
+      //get the query params
+
+      const {
+        q = 0,
```

Comment:
what is the meaning of "q" ?

Reply:
q is the query, set q at description query for search per description.